marshaling.go

```diff
@@ -15,22 +17,38 @@ func Marshal(e *Error) *pe.Error {
 		}
 	}
 
+	// Build message with all the context
+	errMessage := strings.Builder{}
+	errMessage.WriteString(e.Message)
+	next := e.cause
+	for next != nil {
+		errMessage.WriteString(": ")
+		switch typed := next.(type) {
+		case *Error:
+			errMessage.WriteString(typed.Message)
+			next = typed.cause
+		case error:
+			errMessage.WriteString(typed.Error())
+			next = nil
+		}
+	}
+
 	retryable := &pe.BoolValue{}
 	if e.IsRetryable != nil {
 		retryable.Value = *e.IsRetryable
 	}
 
-	err := &pe.Error{
+	err := pe.Error{
 		Code:      e.Code,
-		Message:   e.Message,
+		Message:   errMessage.String(),
```

Comment:
Why not simply call `e.Error()` here rather than mostly duplicate the logic from it above?

Reply:
`e.Error()` includes the code -- `internal_service.code: context: source`